signup-integration-api/src/main/java/io/mosip/signup/api/spi/ProfileRegistryPlugin.java

```diff
@@ -55,5 +58,14 @@ public interface ProfileRegistryPlugin {
      * @return
      */
     boolean isMatch(JsonNode identity, JsonNode inputChallenge);
+
+    /**
+     * Default method to get the UI specification for the registration screen.
+     * @return A map containing the UI specification details.
+     * If not implemented, it throws a custom exception to indicate that the UI specification was not found.
+     */
+    default Map<String, Object> getUISpecification() {
```

Comment:
Instead of taking this as Map and then converting to JsonNode, let us return JsonNode from this method.